sys/posix/sockets/posix_sockets.c

```diff
@@ -170,110 +206,73 @@ static inline int _get_data_from_sockaddr(const struct sockaddr *address, size_t
                 return -1;
             }
             struct sockaddr_in *in_addr = (struct sockaddr_in *)address;
-            *addr = &in_addr->sin_addr;
-            *addr_len = sizeof(ipv4_addr_t);
-            port->u16 = in_addr->sin_port;
+            out->family = AF_INET;
+            out->addr.ipv4 = in_addr->sin_addr.s_addr;
+            out->port = ntohs(in_addr->sin_port);
             break;
+#ifdef SOCK_HAS_IPV6
```

Comment:
Because without `SOCK_HAS_IPV6` defined `sock` does not support IPv6 so without this `#ifdef` the code below would be invalid.